kernel-ewasm/tests/storage_vec.ts

```diff
@@ -87,8 +87,46 @@ describe('StorgeVec', function () {
             assert.strictEqual(length3, 2, "There should be 2 elements");
         });
 
+        it('push and pop third value', async function () {
```

Comment:
For Testing the Lib, it would be preferable to do rust tests locally in the package itself. Was there some issue with `pwasm-test`?

Reply:
Yep, originally I was, but as soon as you involve capabilities you can no longer use `pwasm-test` as it doesn't model multiple contracts and transactions. Once a data structure is able to take a capability it can no longer be tested that way.

Reply:
Hmm. Perhaps one way:
1. In tests, define a custom `Storable` type in tests that simply does a raw `store_write` call during in a `store` call.
1. For `StorageVec::pop` and `StorageVec::push`, you can use the [`cfg!(test)`](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro) macro, and conditionally use a raw `store_write` instead of a syscall `write`.

Reply:
That's feasible, I'd considered doing something like that, but I had to test its function with the capabilities and kernel anyway.